python/python/bystro/tada_mixture/mixture_tada_em.py

```diff
@@ -0,0 +1,394 @@
+
+            eps = 0.001
+            pi_ = pi_ * (1 - K * eps) + np.ones(K) * eps
+
+        self.Lambda = Lambda_
+        self.pi = pi_
+        return self
+
+    def predict(self, data):
+        """
+        This predicts the latent cluster assignments given a fitted model
+
+        Parameters
+        ----------
+        data : np.array-like,shape=(N_variants,n_phenotypes)
+            The data of counts, variants x phenotypes
+
+        Returns
+        -------
+        z_hat : np.array-like,shape=(N_samples,)
+            The cluster identities
+        """
+        log_proba = self.predict_logproba(data)
+        z_hat = np.argmax(log_proba, axis=1)
+        return z_hat
+
+    def predict_logproba(self, data):
+        """
+        Predicts the log probability a datapoint being assigned a specific
+        cluster.
+
+        Parameters
+        ----------
+        data : np.array-like,shape=(N_variants,n_phenotypes)
+            The data of counts, variants x phenotypes
+
+        Returns
+        -------
+        z_hat : np.array-like,shape=(N_samples,)
+            The cluster identities
+        """
+        N = data.shape[0]
+        log_proba = np.zeros((N, self.K))
+        for i in range(N):
+            for k in range(self.K):
+                log_proba[i, k] = np.sum(
+                    st.poisson.logpmf(data[i], self.Lambda[k])
+                )
+        return log_proba
+
+    def _fill_training_options(self, training_options):
+        """
+        This fills any relevant parameters for the learning algorithm
+
+        Parameters
+        ----------
+        training_options : dict
+
+        Returns
+        -------
+        tops : dict
+        """
+        default_options = {"n_iterations": 100}
+        tops = deepcopy(default_options)
+        tops.update(training_options)
+        return tops
+
+
+class MVTadaZPoissonEM(object):
+    def __init__(self, K=4, training_options={}):
+        """
+        This is a Product ZIP latent variable model.
+
+        Parameters
+        ----------
+        K : int,default=4
+            The number of clusters, i.e. number of different types of 
+            genes
+
+        training_options : dict,default={}
+            The parameters for the inference scheme
+        """
+        self.K = int(K)
+        self.training_options = self._fill_training_options(training_options)
+
+    def fit(self, X, progress_bar=True):
+        """
+        Fits a model given count data X.
+
+        Parameters
+        ----------
+        X : np.array-like,shape=(N,p)
+            The count data
+        Returns
+        -------
+        self
+        """
+        td = self.training_options
+        K = self.K
+
+        N, p = X.shape
+
+        model = GaussianMixture(K)
+        model.fit(X)
+        Lambda_list_l = []
+        for i in range(K):
+            Lambda_list_l.append(
+                torch.tensor(
+                    1.2 * model.means_[i].astype(np.float32) + 0.1,
```

Comment:
Could you help me understand this part; should we have 1.2 as a variable / model hyperparameter? I think I understand the 0.1 to be a way of getting away from P(X=0|lambda=0) = 1

Reply:
The reason I initialized it like this was to counteract the zero inflation portion, which is initialized to be non-zero, rough heuristic but there's also substantially less theory in initializing simple models.